tests/xoto3/dynamodb/utils/expressions_test.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+
+from xoto3.dynamodb.utils.expressions import add_variables_to_expression
+
+
+def test_add_variables_to_expression():
+    query_dict = dict()
+    variables = dict(deletedAt="2020-01-01T00:00:00.000000Z")
+
+    add_variables_to_expression(query_dict, variables)
+
+    assert query_dict["ExpressionAttributeNames"] == {"#deletedAt": "deletedAt"}
+    assert query_dict["ExpressionAttributeValues"] == {":deletedAt": "2020-01-01T00:00:00.000000Z"}
+
+
+def test_add_variables_to_expression_with_bad_attribute_name():
+    query_dict = dict()
+    variables = dict(thingy="THINGY", deleted__At="2020-02-02T00:00:00.000000Z")
+
+    with pytest.raises(ValueError, match="Attribute name contains invalid characters"):
+        add_variables_to_expression(query_dict, variables)
+
+
+def test_add_variables_to_expression_with_duplicate_attribute_name():
+    query_dict = dict(ExpressionAttributeNames={"#deletedAt": "deletedAt"},)
+    variables = dict(deletedAt="2020-02-02T00:00:00.000000Z")
+
+    with pytest.raises(
+        ValueError, match="Cannot add a duplicate expression attribute name #deletedAt"
+    ):
+        add_variables_to_expression(query_dict, variables)
+
+
+def test_add_variables_to_expression_with_duplicate_attribute_value():
+    query_dict = dict(ExpressionAttributeValues={":deletedAt": "2020-01-01T00:00:00.000000Z"},)
+    variables = dict(deletedAt="2020-02-02T00:00:00.000000Z")
+
+    with pytest.raises(
+        ValueError, match="Cannot add a duplicate expression attribute value :deletedAt"
```

Comment:
It seems like either of these cases should really raise the same error. Something like `Cannot add a duplicate expression variable 'deletedAt'`

I think I'd be confused by the errors as they are. "Hey... I didn't try to add '#deltedAt'! What are you talking about?!"

And, in practice, would you ever run into one of these without running into the other?

Reply:
No. I can't imagine a single time when you would _normally_ ever encounter the 2nd error without the first one, but if you did, due to some really weird error on your part, knowing which exception took fault with it might be helpful. You're 99% more likely to hit the first one, though...

Reply:
Honestly, though, this was just testing behavior that already existed in the code without any modification to the code itself.